### PR TITLE
Refactor singleton service initializers

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description
Update: Some of the work originally in this PR wound up in the 2.4 release instead. I've now merged master into rel/3.0 and then merged that into this branch. At this point, this PR is some lower priority changes to some service initializers and a lot of test fixture naming convention corrections.

We had some assumptions about the order in which dependencies are initialized, and encountered unrecoverable exceptions when those assumptions were incorrect. Most importantly, accessing `KlaviyoApiClient` before `applicationContext` was available would throw an exception in the initializer. An initialization error cannot be recovered, even if `applicationContext` were later provided, because `init` will only run once. 

# Check List

- [x] Are you changing anything with the public API?
- [x] Are your changes backwards compatible with previous SDK Versions?
- [x] Have you tested this change on real device?
- [x] Have you added unit test coverage for your changes?
- [x] Have you verified that your changes are compatible with all Android versions the SDK currently supports?

## Changelog / Code Overview

As a safety precaution, I've removed all the object initializers that had an implicit dependency, though the API client was the only one I've observed an issue with from testing with our test apps. 

## Test Plan

Updated all the unit tests already
Functional changes are smaller now that some of this branch's original work was cherry picked and released in 2.4 after testing.

## Related Issues/Tickets

CHNL-8569 -- this does not address all the AC of that ticket.

### CODE REVIEW

File: sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt [hunk 1]
```diff
@@ -34,53 +34,20 @@ internal object KlaviyoApiClient : ApiClient {
      */
     private var apiObservers = mutableListOf<ApiObserver>()
 
-    init {
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt:37]: The critical part of this PR: an exception inside an `init` is re-thrown, and besides the initializer doesn't finish properly and can't be re-invoked for a singleton object. Instead, I'm moving these listener startups to a method that we invoke from `Klaviyo.initialize`

File: sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt [hunk 2]
```diff
@@ -61,34 +61,35 @@ internal class KlaviyoTest : BaseTest() {
     }
 
     private val capturedProfile = slot<Profile>()
-    private val apiClientMock: ApiClient = mockk<ApiClient>().apply {
+    private val mockApiClient: ApiClient = mockk<ApiClient>().apply {
```
Comment (evan-masseau) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt:64]: Naming convention fix, sometimes using mock as a prefix, sometimes as a suffix. Suggesting we consolidate to prefix.

File: sdk/core/src/main/java/com/klaviyo/core/KLog.kt [hunk 3]
```diff
@@ -6,7 +6,26 @@ import com.klaviyo.core.config.Log
 import com.klaviyo.core.config.Log.Level
 import java.util.regex.Pattern
 
-open class KLog : Log {
+object KLog : Log {
```
Comment (evan-masseau) [inline sdk/core/src/main/java/com/klaviyo/core/KLog.kt:9]: `KLog` doesn't need to be instantiated, it is essentially just a set of static methods anyway.

File: sdk/core/src/main/java/com/klaviyo/core/lifecycle/KlaviyoLifecycleMonitor.kt [hunk 4]
```diff
@@ -14,10 +14,6 @@ internal object KlaviyoLifecycleMonitor : LifecycleMonitor, Application.Activity
 
     private var activityObservers = mutableListOf<ActivityObserver>()
 
-    init {
-        onActivityEvent { Registry.log.verbose(it.type) }
```
Comment (evan-masseau) [inline sdk/core/src/main/java/com/klaviyo/core/lifecycle/KlaviyoLifecycleMonitor.kt:18]: This one wasn't causing any exceptions, but I thought it best to get rid of any unnecessary object inits to head off the same issues.

**Comment by evan-masseau** [comment]
Reverted to draft, will need to update once 2.4.0 is release. I think these changes were still valuable.

**Review by evan-masseau — COMMENTED** [review]
Merged master into 3.0 and that back into this branch, after the 2.4 release. 
At this point, this PR is some lower priority changes to some service initializers and a lot of test fixture naming convention corrections.

**Review by ajaysubra — APPROVED** [review]

File: sdk/core/src/main/java/com/klaviyo/core/lifecycle/KlaviyoLifecycleMonitor.kt (hunk 4 above)
Comment (dan-peluso) [reply sdk/core/src/main/java/com/klaviyo/core/lifecycle/KlaviyoLifecycleMonitor.kt:21]: Yeah inits can be notoriously hard to track - better safe than sorry

File: versions.properties [hunk 5]
```diff
@@ -11,7 +11,7 @@
 
 # Project versioning, run the following gradle command to update version numbers automatically:
 # ./gradlew bumpVersion --nextVersion=X.Y.Z
-version.klaviyo.versionCode=18
+version.klaviyo.versionCode=19
```
Comment (dan-peluso) [inline versions.properties:14]: Is this going to be the last commit before we cut the release?

**Review by dan-peluso — APPROVED** [review]

File: versions.properties (hunk 5 above)
Comment (evan-masseau) [reply versions.properties:14]: Hm not sure why I had this. I haven't reviewed the 3.0.0 release contents yet, so I'm not certain. We must be getting close though

Comment (evan-masseau) [reply versions.properties:14]:
```suggestion
version.klaviyo.versionCode=18
```
Lets bump this in a standalone PR

**Pull request merged**